core/templates/services/user-backend-api.service.ts

```diff
@@ -0,0 +1,164 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for user data.
+ */
+
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { AppConstants } from 'app.constants';
+import { UserInfo, UserInfoBackendDict } from 'domain/user/user-info.model';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+import { UrlService } from 'services/contextual/url.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+
+interface SubscriptionSummary {
+  'creator_picture_data_url': string;
+  'creator_username': string;
+  'creator_impact': number;
+}
+interface PreferencesBackendDict {
+  'preferred_language_codes': string;
+  'preferred_site_language_code': string;
+  'preferred_audio_language_code': string;
+  'profile_picture_data_url': string;
+  'default_dashboard': string;
+  'user_bio': string;
+  'subject_interests': string;
+  'can_receive_email_updates': boolean;
+  'can_receive_editor_role_email': boolean;
+  'can_receive_feedback_message_email': boolean;
+  'can_receive_subscription_email': boolean;
+  'subscription_list': SubscriptionSummary[];
+}
+interface UrlBackendDict {
+  'login_url': string;
+}
+interface UserContributionRightsDataBackendDict {
+  'can_review_translation_for_language_codes': boolean;
+  'can_review_voiceover_for_language_codes': boolean;
+  'can_review_questions': boolean;
+}
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserBackendApiService {
+  constructor(
+    private http: HttpClient,
+    private urlInterpolationService: UrlInterpolationService,
+    private urlService: UrlService,
+    private windowRef: WindowRef
+  ) {}
+
+    private PREFERENCES_DATA_URL = '/preferenceshandler/data';
+    private USER_CONTRIBUTION_RIGHTS_DATA_URL = '/usercontributionrightsdatahandler'; // eslint-disable-line max-len
+
+    private userContributionRightsInfo = null;
+    private userInfo = null;
+    private returnUrl = '';
+
+    getUserInfoAsync(): Promise<UserInfo> {
+      return new Promise((resolve, reject) => {
+        if (this.urlService.getPathname() === '/signup') {
+          return resolve(UserInfo.createDefault());
+        }
+        if (this.userInfo) {
+          return resolve(this.userInfo);
+        }
+        return this.http.get<UserInfoBackendDict>(
+          '/userinfohandler').toPromise().then(
+          (backendDict) => {
+            if (backendDict.user_is_logged_in) {
```

Comment:
The backend API service should only make HTTP calls, I would suggest leaving all this caching and  check related stuff in the UserService and create a UserBackendAPService which will just have HTTP request functions and use the backend API service in the userService. (I think that's what we expect via issue #8472)

/cc @srijanreddy98, Can you please help @subedi90 to resolve this issue? :)